promql/functions.go

```diff
@@ -298,6 +299,14 @@ func funcSortDesc(ev *evaluator, args Expressions) model.Value {
 	return vector(byValueSorter)
 }
 
+// === sort_by_label(node model.ValVector, model.ValString) model.ValVector ===
+func funcSortByLabel(ev *evaluator, args Expressions) model.Value {
+	l := ev.evalString(args[1])
+	s := vectorByLabelHeap{vector: ev.evalVector(args[0]), labelName: l.String()}
+	sort.Sort(s)
```

Comment:
Thank you for looking at my change. I don't quite follow you though. How is this different from funcSort or funcSortDesc?